File: frontend/src/types/order.ts
export enum OrderSide {
  BUY = 'BUY',
  SELL = 'SELL',
}

export enum OrderType {
  MARKET = 'MARKET',
  LIMIT = 'LIMIT',
}

export enum OrderStatus {
  NEW = 'NEW',
  ACCEPTED = 'ACCEPTED',
  PARTIALLY_FILLED = 'PARTIALLY_FILLED',
  FILLED = 'FILLED',
  CANCELLED = 'CANCELLED',
  REJECTED = 'REJECTED',
  SETTLED = 'SETTLED',
}

export interface Order {
  orderId: string;
  symbol: string;
  side: OrderSide;
  orderType: OrderType;
  quantity: number;
  price?: number;
  status: OrderStatus;
  filledQuantity: number;
  avgFillPrice?: number;
  createdAt: string;
  updatedAt?: string;
  rejectionReason?: string;
}

export interface PlaceOrderRequest {
  symbol: string;
  side: OrderSide;
  orderType: OrderType;
  quantity: number;
  price?: number;
  idempotencyKey: string;
}

export interface OrderResponse {
  orderId: string;
  status: OrderStatus;
  symbol: string;
  side: OrderSide;
  quantity: number;
  price?: number;
  timestamp: string;
}

File: services/OrderIngestion/Models/Order.cs
namespace OrderIngestion.Models;

public enum OrderSide
{
    BUY,
    SELL
}

public enum OrderType
{
    MARKET,
    LIMIT
}

public enum OrderStatus
{
    NEW,
    ACCEPTED,
    REJECTED,
    FILLED,
    CANCELLED,
    SETTLED
}

public class Order
{
    public Guid OrderId { get; set; }
    public string Symbol { get; set; } = string.Empty;
    public OrderSide Side { get; set; }
    public OrderType OrderType { get; set; }
    public decimal Quantity { get; set; }
    public decimal? Price { get; set; }
    public OrderStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? IdempotencyKey { get; set; }
    public string? RejectionReason { get; set; }
}
